source/codegen/service_helpers.py

```diff
@@ -401,19 +401,19 @@ def get_bitfield_value_to_name_mapping(parameter: dict, enums: dict) -> Dict[int
     }
 
 
-def get_resource_handle_type(config: dict) -> str:
-    return config.get("resource_handle_type", "ViSession")
+def get_resource_handle_types(config: dict) -> List[str]:
```

Comment:
await: you need some test coverage for the new codegen. Ideally that would be fake tests and system level tests. But either/or can be OK. In this case "just fake tests" seems like a good option for now. My PR #574 is an OK example of some basic test coverage. We probably don't want to add a lot of "expected failure/noop" system level cases like that, but it's something to consider. And if you have a test system set up you can write tests that can run on the real hw.

Is it possible to test the database without an xnet device?

Reply:
I have added non-ivi fake tests for validating initialize/cleanup when there are multiple session repositories owned by driver. We will probably add more system level tests later once we have enough API coverage ready.